app/controllers/application_controller.rb

```diff
@@ -2,4 +2,14 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+protected
+
+  def authenticate_user!
```

Comment:
You are right!
That method redefinition supposed to be a workaround to some problem and now it is obsolete and not needed anymore.